cmd/export.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cmd defines command line utilities for ghpc
+package cmd
+
+import (
+	"fmt"
+	"hpc-toolkit/pkg/shell"
+	"log"
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	metadataFlag := "metadata"
+	artifactsFlag := "artifacts"
+	exportCmd.Flags().StringVarP(&artifactsDir, artifactsFlag, "a", "", "Alternative artifacts output directory (automatically configured if unset)")
+	exportCmd.Flags().StringVarP(&metadataFile, metadataFlag, "m", "", "Deployment metadata YAML file (automatically configured if unset)")
+	exportCmd.MarkFlagDirname(artifactsFlag)
+	exportCmd.MarkFlagFilename(metadataFlag, "yaml", "yml")
+	rootCmd.AddCommand(exportCmd)
+}
+
+const defaultMetadataFile string = "../.ghpc/deployment_metadata.yaml"
+
+var (
+	artifactsDir string
+	metadataFile string
+	exportCmd    = &cobra.Command{
+		Use:               "export-outputs DEPLOYMENT_DIRECTORY",
+		Aliases:           []string{"output"},
```

Comment:
Consider not providing an alias to start. I feel like the less options we give the simpler the product will be to use. We can always add an option but harder to take away. Also `ghpc output my/dir` just does not seem very clear to me. Thoughts?

What are we planning for the inverse command?

I like the following options:
```
# you could add hyphen, I like that propagate could apply to both to make clear they go together
ghpc propagate outputs my/dir
ghpc propagate inputs my/dir

# This is also clear. I don't expect people to try to remember the commands
ghpc export-outputs my/dir
ghpc collect-inputs my/dir
```

Reply:
I can remove the alias. I have simplified the "import" command to address only the use case where the export command has identified an alternative artifacts directory. In this case, it will look for appropriately named files in the artifacts directory and stage them into the right group directory. If they are not found, import fails.

In other words, our "simple" case for deploying will only require the exports-outputs command.

Context: in a CI/CD pipeline, one is likely to find it easy to chain stages together in which all artifacts are tarballed from stage 0 and copied into a single folder in stage N.